ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -590,6 +590,78 @@ const CreatePanel = ({creatorFreeCreditsRemaining, preset, profile, setPanelCont
   </div>;
 };
 
+const ConfirmUpdatePanel = ({initialRuntimeConfig, newRuntimeConfig, onCancel, updateButton}) => {
+  const runtimeDiffs = getRuntimeConfigDiffs(initialRuntimeConfig, newRuntimeConfig);
+  const needsDelete = runtimeDiffs.map(diff => diff.differenceType).includes(RuntimeDiffState.NEEDS_DELETE);
+
+  return <React.Fragment>
+    <div style={styles.controlSection}>
+      <h3 style={{...styles.baseHeader, ...styles.sectionHeader, marginTop: '.1rem', marginBottom: '.2rem'}}>Editing your environment</h3>
+      <div>
+        You're about to apply the following changes to your environment:
+      </div>
+      <ul>
+        {runtimeDiffs.map(diff =>
+          <li>
+            {diff.desc} from <b>{diff.previous}</b> to <b>{diff.new}</b>
+          </li>
+        )}
+      </ul>
+      <FlexRow style={{marginTop: '.5rem'}}>
+        <div style={{marginRight: '1rem'}}>
+          <b style={{fontSize: 10}}>New estimated cost</b>
+          <div style={{...styles.costPredictor, padding: '.25rem .5rem'}}>
+            <CostEstimator runtimeParameters={newRuntimeConfig}/>
+          </div>
+        </div>
+        <div>
+          <b style={{fontSize: 10}}>Previous estimated cost</b>
+          <div style={{...styles.costPredictor,
+            padding: '.25rem .5rem',
+            color: 'grey',
+            backgroundColor: ''}}>
+            <CostEstimator runtimeParameters={initialRuntimeConfig} costTextColor='grey'/>
+          </div>
+        </div>
+      </FlexRow>
+    </div>
+
+    <WarningMessage>
+      <TextColumn>
+        {needsDelete ? <React.Fragment>
+          <div>
+            You've made changes that can only take effect upon deletion and re-creation of
+            your cloud environment.
+          </div>
+          <div style={{marginTop: '0.5rem'}}>
+            Any in-memory state and local file modifications will be erased. Data stored in
+            workspace buckets is never affected by changes to your cloud environment.
+          </div>
+        </React.Fragment> : <React.Fragment>
+          <div>
+            These changes require a reboot of your environment to take effect.
+          </div>
+          <div style={{marginTop: '0.5rem'}}>
+            Any in-memory state will be erased, but local file modifications will be preserved.
+            Data stored in workspace buckets is never affected by changes to your cloud environment.
+          </div>
+        </React.Fragment>}
+      </TextColumn>
+    </WarningMessage>
+
+    <FlexRow style={{justifyContent: 'flex-end', marginTop: '.75rem'}}>
+      <Button
+        type='secondary'
+        aria-label='Cancel'
+        style={{marginRight: '.25rem'}}
+        onClick={onCancel}>
+        Cancel
+      </Button>
+      {updateButton}
```

Comment:
In the mocks, the text of the update button is different in each case. Fine if this gets tracked in the follow-up ticket though.